File: backend/src/main/java/com/hruthvik/jobtracker/service/AuthService.java
package com.hruthvik.jobtracker.service;

import com.hruthvik.jobtracker.dto.request.AuthRequest;
import com.hruthvik.jobtracker.dto.request.RegisterRequest;
import com.hruthvik.jobtracker.dto.response.AuthResponse;
import com.hruthvik.jobtracker.dto.response.UserResponse;
import com.hruthvik.jobtracker.entity.User;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse authenticate(AuthRequest request);
    User getCurrentUser();
    UserResponse getMe();
}

File: frontend/src/services/auth.service.ts
import api from './api';
import { AuthResponse, User } from '../types';

export const authService = {
  login: async (data: any): Promise<AuthResponse> => {
    const response = await api.post('/auth/login', data);
    return response.data;
  },
  register: async (data: any): Promise<AuthResponse> => {
    const response = await api.post('/auth/register', data);
    return response.data;
  },
  getMe: async (): Promise<User> => {
    const response = await api.get('/auth/me');
    return response.data;
  },
};
